src/types/Typechecker/TypeError.hs

```diff
@@ -391,10 +391,11 @@ instance Show Error where
         printf
           "Conjunctive traits '%s' and '%s' cannot share mutable field '%s'"
            (show left) (show right) (show field)
-    show (OverriddenMethodTypeError name expected trait) =
+    show (OverriddenMethodTypeError name expected trait actual) =
         printf ("Overridden method '%s' does not " ++
-                "have the expected type '%s' required by %s")
-               (show name) (show expected) (refTypeName trait)
+                "have the expected type '%s' required by %s.\n" ++
```

Comment:
It seems we do not have a test for this! This PR does not introduce this error message (it only improves it), but I will add a test or two here anyway.